feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c

```diff
@@ -300,24 +307,26 @@ bool target_radio_config_set2(const struct schema_Wifi_Radio_Config *rconf,
 		int tx_ant_avail;
 		tx_ant_avail=phy_get_tx_available_antenna(phy);
 
-		if ((rconf->tx_chainmask & tx_ant_avail) != rconf->tx_chainmask) {
+		if (rconf->tx_chainmask == 0) {
 			blobmsg_add_u32(&b, "txantenna", tx_ant_avail);
-			LOGN("maximum number of tx antennae available to use:%d",tx_ant_avail);
-		}
-		else
+		} else if ((rconf->tx_chainmask & tx_ant_avail) != rconf->tx_chainmask) {
+			blobmsg_add_u32(&b, "txantenna", tx_ant_avail);
+		} else {
 			blobmsg_add_u32(&b, "txantenna", rconf->tx_chainmask);
+		}
 	}
 
 	if (changed->rx_chainmask) {
 		int rx_ant_avail;
 		rx_ant_avail=phy_get_rx_available_antenna(phy);
 
-		if ((rconf->rx_chainmask & rx_ant_avail) != rconf->rx_chainmask) {
+		if (rconf->rx_chainmask == 0) {
 			blobmsg_add_u32(&b, "rxantenna", rx_ant_avail);
-			LOGN("maximum number of rx antennae available to use:%d",rx_ant_avail);
-		}
-		else
+		} else if ((rconf->rx_chainmask & rx_ant_avail) != rconf->rx_chainmask) {
+			blobmsg_add_u32(&b, "rxantenna", rx_ant_avail);
+		} else {
 			blobmsg_add_u32(&b, "rxantenna", rconf->rx_chainmask);
```

Comment:
Wireless UCI page says if the parameter "diversity" is enabled, the parameter rxantenna has no effect and "diversity" is enabled by default.

Reply:
Okay, "diversity" is not read by hostapd. So I think we are good here.